source/Distribution.cpp

```diff
@@ -0,0 +1,75 @@
+/* Distribution.cpp
+Copyright (c) 2022 by DeBlister
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+*/
+
+#include "Distribution.h"
+
+#include "Random.h"
+
+#include <vector>
+
+namespace {
+	double ManipulateNormal(double smoothness, bool invert)
+	{
+		// Center values within [0, 1] so that fractional retention begins to accumulate
+		// at the endpoints (rather than at the center) of the distribution.
+		double randomFactor = Random::BMNormal(0.5, smoothness);
+		// Retain only the fractional information, to keep all absolute values within [0, 1].
+		// Might be possible to get away with int32_t here, not sure.
+		randomFactor = randomFactor - static_cast<int64_t>(randomFactor);
+
+		// Shift negative values into [0, 1] to create redundancy at the endpoints
+		if(randomFactor < 0.)
+			randomFactor++;
+
+		// Invert probabilities so that endpoints are most probable.
+		if(invert)
+		{
+			if(randomFactor > 0.5)
+				randomFactor -= 0.5;
+			else if(randomFactor < 0.5)
+				randomFactor += 0.5;
+		}
+
+		// Transform from [0, 1] to [-1, 1] so that the return value can be simply used.
+		randomFactor *= 2;
+		randomFactor--;
+
+		return randomFactor;
+	}
+
+	// These values are paired with Distribution::Types; any change in one should be made in the other.
+	const std::vector<double> SMOOTHNESS_TABLE = { 0.13, 0.234, 0.314 };
```

Comment:
```suggestion
	const double SMOOTHNESS_TABLE[] = {.13, .234, .314};
```

Reply:
This would make it the only array (new or old) in the entire code, I think.  I could use std::array instead and and see what reaction there is to that if you just don't think there should be a vector here, but I don't think the old style array is going to fly.

Reply:
I went back and looked and boy was I wrong.  Glad you brought this up.